lib/src/test/java/io/ably/lib/test/realtime/RealtimeMessageTest.java

```diff
@@ -1010,4 +1014,42 @@ public void should_have_serial_action_createdAt() throws AblyException {
             assertNull(msgComplete.waitFor(1, 10_000));
         }
     }
+
+    @Test
+    public void should_not_duplicate_messages() throws Exception {
+        ClientOptions opts = createOptions(testVars.keys[0].keyStr);
+        opts.logLevel = 2;
+        String testChannelName = "my-channel" + System.currentTimeMillis();
+        try (AblyRest rest = new AblyRest(opts)) {
+            final io.ably.lib.rest.Channel channel = rest.channels.get(testChannelName);
+            List<String> testMessagesData = List.of("message 1", "message 2", "message 3");
+            List<Message> messages = testMessagesData
+                .stream()
+                .map(data -> new Message("name", data))
+                .collect(Collectors.toList());
+            channel.publish(messages.toArray(Message[]::new));
+        }
+
+        try (AblyRealtime realtime = new AblyRealtime(opts)) {
+            final ChannelOptions options = new ChannelOptions();
+            options.params = Map.of("rewind", "10");
+            final Channel channel = realtime.channels.get(testChannelName, options);
+            final CompletionWaiter completionWaiter = new CompletionWaiter();
+            final AtomicInteger counter = new AtomicInteger();
```

Comment:
Also, we can avoid using `AtomicInteger` since the internal `emit` call for channel messages is blocking in nature and the next event will only be sent when the previous one is processed

Reply:
Seems, IDE suggests to make it either atomic or final : (

Reply:
I think test can be further simplified to 
```
        ClientOptions opts = createOptions(testVars.keys[0].keyStr);
        opts.logLevel = 2;
        String testChannelName = "my-channel" + System.currentTimeMillis();
        try (AblyRest rest = new AblyRest(opts)) {
            final io.ably.lib.rest.Channel channel = rest.channels.get(testChannelName);
            Message[] messages = new Message[3];
            for (int i = 0; i < messages.length; i++) {
                messages[i] = new Message("name", "message" + i);
            }
            channel.publish(messages);
        }

        try (AblyRealtime realtime = new AblyRealtime(opts)) {
            final ChannelOptions options = new ChannelOptions();
            options.params = Map.of("rewind", "10");
            final Channel channel = realtime.channels.get(testChannelName, options);

            final int[] receivedMessages = {0};
            channel.subscribe((Channel.MessageListener) message -> receivedMessages[0] = receivedMessages[0] + 1);

            Exception conditionError = new Helpers.ConditionalWaiter().wait(() -> receivedMessages[0] == 3, 10000);
            assertNull(conditionError);

            assertEquals("Should be exactly 3 messages",3, receivedMessages[0]);

            Thread.sleep(1500);

            assertEquals("Should be exactly 3 messages even after 1.5 sec wait",3, receivedMessages[0]);
```
`ConditionalWaiter` is same as `assertWaiter`

Reply:
it's a test, we just need wrapper around counter :) we don't care about atomic's overhead here (it's probably less than millisecond). Overhead from our `CompletionWaiter` are incomparable to it

Reply:
IDE is saying, `messages.toArray(Message[]::new)` is only compatible with java 11+

Reply:
once again it's a test, this code will be never executed by library users we still target 1.8

Reply:
Yeah, but not sure why IDE is signaling warning for use of `List.of` and `Map.of` ( java 9+ compatible )

Reply:
I think we can ignore those warnings, but shouldn't change IDE settings for the same.
Doing so can introduce java 8+ code in the source code as well.

Reply:
I rewrote to use 1.8, to get rid of IDE warnings